### DISCUSSION TITLE
What configuration values are the best for INMP441 microphone

### DISCUSSION THREAD

**Discussion by wojciechczyz** [stas-sl/esphome-sound-level-meter#6] [Q&A]
I have connected INMP441 as suggested in code:

BoardLolin32 lite | Microphone INMP441 | Code
-- | -- | --
GND | GND |  
3V | VDD |  
19 | SD | din_pin
18 | WS | ws_pin
23 | SCK | bck_pin

What other values (based on advanced configuration) should I set?
Well to give specific problem - when I use configuration, I am getting baseline reading of ~53 dB. My room is very quiet.
I am using INMP441 so do I need to just offset this or am I missing something? L/R pin is left with no connection.

Thank you,
Wojciech

**Comment by stas-sl** [comment]
I'm using INMP441 myself, so the settings in the example config should work for it as is. 

There could be multiple reasons of having high readings. Do you mean 53 dBA, i.e. having A-weighting applied? As dBZ values (no weighting) are usually 5-10 dB higher. Do they increase reasonably if you expose it to some louder sounds? 

It might be tricky to investigate the cause of extra noise. I spent quite a bit of time myself doing this.  Some simple ideas to try are:
1. Connect L/R to GND.
2. Turn off WiFI on ESP32. (I noticed that for some boards WiFi causes a lot of interference, while for other there is no such issue at all)
3. Move the mic around the board, closer/farther from it.
4. Try different PINS for GND/3V
5. Maybe try different boards?

If nothing from above helps, I'm afraid the last option would be to write a sketch that records that noise (whether to a file on a SD card or streams it via WiFi/UART) and then analyze/listen it on PC, though that is not an easy adventure.

According to the datasheet INMP441 has 33 dBA self noise, however I've never seen values lower than ~36 dBA with my mics. So that is what you should expect, if you'll be able to eliminate extra noise.
